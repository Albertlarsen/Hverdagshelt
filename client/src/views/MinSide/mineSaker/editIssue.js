// @flow

import React from "react";
import { Button, Form, FormControl } from "react-bootstrap";
import { Issue } from "../../../classTypes";
import { IssueService } from "../../../services";
import NavLink from "react-router-dom/es/NavLink";
import Nav from "react-bootstrap/es/Nav";
import NavItem from "react-bootstrap/es/NavItem";
import Progress from "reactstrap/es/Progress";
import { PageHeader } from "../../../components/PageHeader/PageHeader";
import { history } from "../../../index";

let issueService = new IssueService();

export class EditIssue extends React.Component<{
  match: { params: { issueId: number } }
}> {
  state = {
    text: ""
  };

  componentDidMount() {
    console.log(this.props.match.params.issueId);
    issueService
      .getIssueAndCounty(this.props.match.params.issueId)
      .then(res => {
        this.setState({
          text: res[0].text
        });
      });
  }

  handleChangeText = () => (event: SyntheticInputEvent<HTMLInputElement>) => {
    this.setState({
      text: event.target.value
    });
  };

  render() {
    console.log(this.state.text);
    return (
      <div className="container">
        <Form>
          <FormControl
            style={{ margin: "5px" }}
            value={this.state.text}
            componentClass="textarea"
            onChange={this.handleChangeText()}
          />
          <Button
            bsStyle="primary"
            onClick={this.save}
            style={{ margin: "5px" }}
          >
            Lagre
          </Button>
        </Form>
      </div>
    );
  }

  save = () => {
    console.log(this.props.match.params.issueId);
    console.log(this.state.text);
    issueService.editIssue(this.props.match.params.issueId, this.state.text);
    history.push("/min_side/mine_saker");
    window.location.reload();
  };
}
