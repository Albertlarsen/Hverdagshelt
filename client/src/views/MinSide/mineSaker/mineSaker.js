// @flow

import React from 'react';
import {Grid, Col, Row, Button, Table} from "react-bootstrap";
import {Issue} from "../../../classTypes";
import {UserService} from "../../../services";
import {Alert} from "../../../widgets";
import ProgressBar from "react-bootstrap/es/ProgressBar";
import {Status} from "../../../classTypes";
import NavLink from "react-router-dom/es/NavLink";
import Nav from "react-bootstrap/es/Nav";
import NavItem from "react-bootstrap/es/NavItem";

let jwt = require("jsonwebtoken");
let userService = new UserService();

let status = null;

interface State{
    issues: Object[];
    category: Object[];
    decoded: Object;
}//end interface

interface Props{}

export class MineSaker extends React.Component<Props,State>{
    state = {
        issues: [],
        category: [],
        decoded: jwt.verify(window.localStorage.getItem('userToken'), "shhhhhverysecret")
    };
    render(){
        return(
            <Grid>
                <br/>
                <br/>
            <Table>
                <thead>
                    <tr>
                        <th>
                            Beskrivelse
                        </th>
                        <th>
                            Kategori
                        </th>
                        <th>
                            Status
                        </th>
                    </tr>
                </thead>
                <tbody>
                {this.state.issues.map(e => {
                    return(
                        <tr key={e.text}>
                            <td>
                                <Nav bsStyle="pills">
                                    <NavItem href={"/#min_side/sakoversikt/" + this.state.decoded.email + "/" + e.issueId}>
                                    {e.text}
                                    </NavItem>
                                </Nav>
                            </td>
                            <td>
                                {e.category}
                            </td>
                            <td>
                                {this.updateStatus(e.statusName)}
                                <ProgressBar bsStyle={this.status.progressBar} now={this.status.progress}
                                label={e.statusName}/>
                            </td>
                        </tr>
                    )
                })}
                </tbody>
            </Table>
            </Grid>
        )
    }//end method

    componentDidMount(){
        let decoded = jwt.verify(window.localStorage.getItem('userToken'), "shhhhhverysecret");
        userService.getMyIssues(decoded.email).then(response => {
            this.setState({issues: response});
        }).catch((error: Error) => Alert.danger(error.message));
    }//end method

    //to set progressbar
    updateStatus(status: string){
        this.status = new Status(status);
    }//end method

}//end class