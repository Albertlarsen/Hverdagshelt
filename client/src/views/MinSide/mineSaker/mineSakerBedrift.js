// @flow

import React from 'react';
import {Grid, Col, Row, Button, Table} from "react-bootstrap";
import {Issue} from "../../../classTypes";
import {CategoryService, UserService} from "../../../services";
import {Alert} from "../../../widgets";
import ProgressBar from "react-bootstrap/es/ProgressBar";
import {Status} from "../../../classTypes";
import NavLink from "react-router-dom/es/NavLink";
import Nav from "react-bootstrap/es/Nav";
import NavItem from "react-bootstrap/es/NavItem";
import Progress from "reactstrap/es/Progress";

let jwt = require("jsonwebtoken");
let userService = new UserService();
let categoryService = new CategoryService();

interface State{
    issues: Object[];
    decoded: Object;
}

export class mineSakerBedrift extends React.Component<State>{
    state = {
      issues: [],
      decoded: jwt.verify(window.localStorage.getItem('userToken'), "shhhhhverysecret")
    };

    componentWillMount() {
        userService.getCompanyIssues(this.state.decoded.email).then(response => {
            this.setState({issues: response});
            this.getSorted();
        }).catch((error: Error) => Alert.danger(error.message));
    }//end method

    render(){
        return(
            <Grid>
                <br/>
                <br/>
                <Table>
                    <thead>
                    <tr>
                        <th>
                            Beskrivelse
                        </th>
                        <th>
                            Adresse
                        </th>
                        <th>
                            Status
                        </th>
                    </tr>
                    </thead>
                    <tbody>
                    {this.state.issues.map(e => {
                        return(
                            <tr key={e.issueId}>
                                <td>
                                    <Nav bsStyle="pills">
                                        <NavItem href={"/#min_side/sakoversikt/" + this.state.decoded.email + "/" + e.issueId}>
                                            {e.text}
                                        </NavItem>
                                    </Nav>
                                </td>
                                <td>
                                    {e.address}
                                </td>
                                <td>
                                    {this.updateStatus(e.statusName)}
                                    <ProgressBar>
                                    <ProgressBar bsStyle={this.status.progressBar} active now={this.status.progress}
                                             label={e.statusName} style={{color: 'black'}} key={1}/>
                                    </ProgressBar>
                                </td>
                            </tr>
                        )
                    })}
                    </tbody>
                </Table>
            </Grid>
        )
    }//end method

    //To set progressbar
    updateStatus(status: string){
        this.status = new Status(status);
    }//end method

    getSorted = () => {
        //Sorting view so completed issues are listed at the bottom
        let sorted: Object = [];
        this.state.issues.map(e => {
            if(e.statusName == 'Registered'){
                sorted.push(e)
            }
        });
        this.state.issues.map(e => {
            if(e.statusName === 'In progress'){
                sorted.push(e)
            }
        });
        this.state.issues.map(e => {
            if(e.statusName === 'Completed'){
                sorted.push(e)
            }
        });
        this.setState({issues: sorted});
    };//end method
}//end class