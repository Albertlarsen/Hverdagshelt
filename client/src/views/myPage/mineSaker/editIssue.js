// @flow

import React from "react";
import { Button, Form, FormControl } from "react-bootstrap";
import { Issue } from "../../../classTypes";
import { IssueService } from "../../../services";
import { history } from "../../../index";

let issueService = new IssueService();

export class EditIssue extends React.Component<{
  match: { params: { issueId: number } }
}> {
  state = {
    text: ""
  };

  componentDidMount() {
    issueService
      .getIssueAndCounty(this.props.match.params.issueId)
      .then(res => {
        this.setState({
          text: res[0].text
        });
      });
  }

  handleChangeText = () => (event: SyntheticInputEvent<HTMLInputElement>) => {
    this.setState({
      text: event.target.value
    });
  };

  render() {
    return (
      <div className="container">
        <Form>
          <FormControl
            style={{ margin: "5px" }}
            value={this.state.text}
            componentClass="textarea"
            onChange={this.handleChangeText()}
          />
          <Button
            bsStyle="primary"
            onClick={this.save}
            style={{ margin: "5px" }}
          >
            Lagre
          </Button>
        </Form>
      </div>
    );
  }

  save = () => {
    issueService.editIssue(this.props.match.params.issueId, this.state.text);
    window.location.reload();
  };


}
