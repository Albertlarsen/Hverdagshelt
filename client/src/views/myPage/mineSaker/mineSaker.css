.knapp {
  margin-left: 10px;
}
