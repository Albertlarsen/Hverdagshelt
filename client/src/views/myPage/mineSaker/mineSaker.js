// @flow

import React from 'react';
import {
  Grid,
  Button,
  Table,
  ProgressBar,
  Nav,
  NavItem,
  ToggleButton,
  ToggleButtonGroup, FormControl,
} from 'react-bootstrap';
import { Issue } from '../../../classTypes';
import { CategoryService, IssueService, UserService } from '../../../services';
import { Filter } from '../../../components/Filter/Filter';
import { Alert } from '../../../widgets';
import { Status } from '../../../classTypes';
import { PageHeader } from '../../../components/PageHeader/PageHeader';
import { history } from '../../../index';
import mineSaker from './mineSaker.css';
import update from "immutability-helper";

let jwt = require('jsonwebtoken');
let userService = new UserService();
let categoryService = new CategoryService();
let issueService = new IssueService();
let filter = new Filter();
let buffer = {};

interface State {
  issues: Object[];
  category: Object;
  category1: Object[];
  category2: Object[];
  category3: Object[];
  edit: Object;
}//end interface

interface Props {
}

/**
 * @class MineSaker
 */
export class MineSaker extends React.Component<Props, State> {
  match: { params: { mail: string } };
  constructor(props){
    super(props);
    this.state = {
      issues: [],
      category: [],
      category1: [],
      category2: [],
      category3: [],
      edit: {}
    };
  }//end constructor

    /**
     * deletes an event from user's list of events registered
     *
     * @method delete
     * @param issueId
     * @param statusName
     * @returns void
     */
  delete(issueId: number, statusName: string) {
    if (statusName == 'Registered') {
      if (confirm('Er du sikker på at du vil slette denne saken?')) {
        issueService.deleteIssue(issueId);
        window.location.reload();
      }
    } else if (statusName == 'In progress') {
      alert('denne saken er under arbeid, og kan ikke slettes');
    } else {
      alert('Du kan ikke slette ferdige saker');
    }
    console.log(issueId);
    console.log(statusName);
    //issueService.deleteIssue(issueId);
  }

  componentWillMount() {
    userService.getMyIssues().then(response => {
      this.setState({ issues: response }, () => this.getSorted());
    }).catch((error: Error) => Alert.danger(error.message));

    categoryService.getCategory1().then(response => {
      this.setState({ category1: response });
    }).catch((error: Error) => Alert.danger(error.message));

    categoryService.getCategory2().then(response => {
      this.setState({ category2: response });
    }).catch((error: Error) => Alert.danger(error.message));

    categoryService.getCategory3().then(response => {
      this.setState({ category3: response });
    }).catch((error: Error) => Alert.danger(error.message));
  }//end method

  render() {
    let cat = [];
    this.state.issues.map(issue => {
      if (issue.categoryLevel === 1) {
        this.category = this.state.category1.find(e => e.categoryId === issue.categoryId);
        cat.push(this.category);
      } else if (issue.categoryLevel === 2) {
        this.category = this.state.category2.find(e => e.category2Id === issue.categoryId);
        cat.push(this.category);
      } else {
        this.category = this.state.category3.find(e => e.category3Id === issue.categoryId);
        cat.push(this.category);
      }//end condition
    });
    return (

      <div className="bottomFooter">
        <Grid>
          <PageHeader title={'Mine saker'}/>
          <div align="center">
            <ToggleButtonGroup type="radio" name="options" defaultValue={1}>
              <ToggleButton onClick={filter.showAll} bsStyle={'primary'} value={1}>Alle</ToggleButton>
              <ToggleButton onClick={filter.filterRegistered} bsStyle={'primary'} value={2}>Registrerte</ToggleButton>
              <ToggleButton onClick={filter.filterInProgress} bsStyle={'primary'} value={3}>Behandles</ToggleButton>
              <ToggleButton onClick={filter.filterCompleted} bsStyle={'primary'} value={4}>Fullført</ToggleButton>
            </ToggleButtonGroup>
          </div>
          <Table id={'myTable'}>
            <thead>
            <tr>
              <th>Beskrivelse</th>
              <th>Kategori</th>
              <th>Status</th>
              <th>Endre</th>
            </tr>
            </thead>
            <tbody>
            {this.state.issues.map((e,i) => {
              if(e.issueId === this.state.edit) {
                return (
                    <tr key={e.issueId}>
                      <td>
                        <FormControl componentClass={"textarea"}
                                     value={e.text}
                                     onChange={this.handleTextChange(i)}/>
                      </td>
                      <td>{this.setCategory(cat, i)}</td>
                      <td>
                        {this.updateStatus(e.statusName)}
                        <ProgressBar>
                          <ProgressBar
                              bsStyle={this.status.progressBar}
                              active={this.status.inProgress}
                              now={this.status.progress}
                              label={this.status.name}
                              style={{color: 'black'}}
                              key={1}
                          />
                        </ProgressBar>
                      </td>
                      <td>

                        <Button className="knapp" bsStyle="primary"
                                onClick={() => this.save(e)}>
                          Lagre
                        </Button>

                        <Button className="knapp" bsStyle="link" style={{color: 'darkred'}}
                                onClick={() => this.cancel()}>
                          Avbryt
                        </Button>
                      </td>
                    </tr>
                )
              }else{
                return (
                    <tr key={e.text}>
                      <td style={{'max-width':'10px'}}>
                        <Nav bsStyle="pills">
                          <NavItem href={'/#min_side/sakoversikt/' + e.issueId}>
                            {e.text}
                          </NavItem>
                        </Nav>
                      </td>
                      <td>{this.setCategory(cat, i)}</td>
                      <td>
                        {this.updateStatus(e.statusName)}
                        <ProgressBar>
                          <ProgressBar
                              bsStyle={this.status.progressBar}
                              active={this.status.inProgress}
                              now={this.status.progress}
                              label={this.status.name}
                              style={{color: 'black'}}
                              key={1}
                          />
                        </ProgressBar>
                      </td>
                      <td>

                        <Button className="knapp" bsStyle="link"
                                onClick={() => this.editNow(e.issueId)}>
                          <span className="glyphicon glyphicon-pencil"></span>
                        </Button>

                        <Button className="knapp" bsStyle="link" style={{color: 'darkred'}}
                                onClick={this.delete.bind(
                                    this,
                                    e.issueId,
                                    e.statusName
                                )}>
                          <span className="glyphicon glyphicon-trash"></span>
                        </Button>
                      </td>
                    </tr>
                )
              }//end method
            })}
            </tbody>
          </Table>
        </Grid>
      </div>
    );
  }//end method


  //To set progressbar
  updateStatus(status: string) {
    this.status = new Status(status);
  }//end method

  getSorted = () => {
    //Sorting view so completed issues are listed at the bottom
    let sorted: Object = [];
    this.state.issues.map(e => {
      if (e.statusName === 'Registered') {
        sorted.push(e);
      }
    });
    this.state.issues.map(e => {
      if (e.statusName === 'In progress') {
        sorted.push(e);
      }
    });
    this.state.issues.map(e => {
      if (e.statusName === 'Completed') {
        sorted.push(e);
      }
    });
    this.setState({ issues: sorted }, () => buffer = this.state.issues);
  };//end method

  setCategory = (cat: Object[], i: number) => {
    if (cat[i] !== undefined)
      return <div> {cat[i].name}</div>;
  };//end method

  editNow(issueId: number){
    this.setState({edit: issueId});
  }//end method

  save = (issue: Object) => {
    issueService.editIssue(issue.issueId, issue.text);
    this.setState({edit: {}})
  };

  cancel(){
    this.setState({edit: {}, issues: buffer})
  }//end method

  handleTextChange = (index: number) => (event: Event) => {
    this.setState({issues: update(this.state.issues, {[index]: {text: {$set: event.target.value}}})})
  };

}//end class