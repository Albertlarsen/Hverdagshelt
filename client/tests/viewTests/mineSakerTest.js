// @flow

import {shallow} from "enzyme/build";
import * as React from "react";
import {MineSaker} from "../../src/views/MinSide/mineSaker/mineSaker";

describe('Test for MineSaker view', () => {
    const wrapper = shallow(<MineSaker />);

    it('renders correctly', () => {
        expect(wrapper).toMatchSnapshot();
    });
});