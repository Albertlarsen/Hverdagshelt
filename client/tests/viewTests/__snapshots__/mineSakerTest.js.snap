// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test for MineSaker view renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <MineSaker />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <Grid
        bsClass="container"
        componentClass="div"
        fluid={false}
      >
        <PageHeader
          title="Mine saker"
        />
        <Table
          bordered={false}
          bsClass="table"
          condensed={false}
          hover={false}
          responsive={false}
          striped={false}
        >
          <thead>
            <tr>
              <th>
                Beskrivelse
              </th>
              <th>
                Kategori
              </th>
              <th>
                Status
              </th>
              <th>
                Endre
              </th>
            </tr>
          </thead>
          <tbody />
        </Table>
      </Grid>,
      "className": "bottomFooter",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "bsClass": "container",
        "children": Array [
          <PageHeader
            title="Mine saker"
          />,
          <Table
            bordered={false}
            bsClass="table"
            condensed={false}
            hover={false}
            responsive={false}
            striped={false}
          >
            <thead>
              <tr>
                <th>
                  Beskrivelse
                </th>
                <th>
                  Kategori
                </th>
                <th>
                  Status
                </th>
                <th>
                  Endre
                </th>
              </tr>
            </thead>
            <tbody />
          </Table>,
        ],
        "componentClass": "div",
        "fluid": false,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "title": "Mine saker",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "bordered": false,
            "bsClass": "table",
            "children": Array [
              <thead>
                <tr>
                  <th>
                    Beskrivelse
                  </th>
                  <th>
                    Kategori
                  </th>
                  <th>
                    Status
                  </th>
                  <th>
                    Endre
                  </th>
                </tr>
              </thead>,
              <tbody />,
            ],
            "condensed": false,
            "hover": false,
            "responsive": false,
            "striped": false,
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <tr>
                  <th>
                    Beskrivelse
                  </th>
                  <th>
                    Kategori
                  </th>
                  <th>
                    Status
                  </th>
                  <th>
                    Endre
                  </th>
                </tr>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <th>
                      Beskrivelse
                    </th>,
                    <th>
                      Kategori
                    </th>,
                    <th>
                      Status
                    </th>,
                    <th>
                      Endre
                    </th>,
                  ],
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Beskrivelse",
                    },
                    "ref": null,
                    "rendered": "Beskrivelse",
                    "type": "th",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Kategori",
                    },
                    "ref": null,
                    "rendered": "Kategori",
                    "type": "th",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Status",
                    },
                    "ref": null,
                    "rendered": "Status",
                    "type": "th",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Endre",
                    },
                    "ref": null,
                    "rendered": "Endre",
                    "type": "th",
                  },
                ],
                "type": "tr",
              },
              "type": "thead",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [],
              },
              "ref": null,
              "rendered": Array [],
              "type": "tbody",
            },
          ],
          "type": [Function],
        },
      ],
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <Grid
          bsClass="container"
          componentClass="div"
          fluid={false}
        >
          <PageHeader
            title="Mine saker"
          />
          <Table
            bordered={false}
            bsClass="table"
            condensed={false}
            hover={false}
            responsive={false}
            striped={false}
          >
            <thead>
              <tr>
                <th>
                  Beskrivelse
                </th>
                <th>
                  Kategori
                </th>
                <th>
                  Status
                </th>
                <th>
                  Endre
                </th>
              </tr>
            </thead>
            <tbody />
          </Table>
        </Grid>,
        "className": "bottomFooter",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "bsClass": "container",
          "children": Array [
            <PageHeader
              title="Mine saker"
            />,
            <Table
              bordered={false}
              bsClass="table"
              condensed={false}
              hover={false}
              responsive={false}
              striped={false}
            >
              <thead>
                <tr>
                  <th>
                    Beskrivelse
                  </th>
                  <th>
                    Kategori
                  </th>
                  <th>
                    Status
                  </th>
                  <th>
                    Endre
                  </th>
                </tr>
              </thead>
              <tbody />
            </Table>,
          ],
          "componentClass": "div",
          "fluid": false,
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "title": "Mine saker",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "bordered": false,
              "bsClass": "table",
              "children": Array [
                <thead>
                  <tr>
                    <th>
                      Beskrivelse
                    </th>
                    <th>
                      Kategori
                    </th>
                    <th>
                      Status
                    </th>
                    <th>
                      Endre
                    </th>
                  </tr>
                </thead>,
                <tbody />,
              ],
              "condensed": false,
              "hover": false,
              "responsive": false,
              "striped": false,
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <tr>
                    <th>
                      Beskrivelse
                    </th>
                    <th>
                      Kategori
                    </th>
                    <th>
                      Status
                    </th>
                    <th>
                      Endre
                    </th>
                  </tr>,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      <th>
                        Beskrivelse
                      </th>,
                      <th>
                        Kategori
                      </th>,
                      <th>
                        Status
                      </th>,
                      <th>
                        Endre
                      </th>,
                    ],
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": "Beskrivelse",
                      },
                      "ref": null,
                      "rendered": "Beskrivelse",
                      "type": "th",
                    },
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": "Kategori",
                      },
                      "ref": null,
                      "rendered": "Kategori",
                      "type": "th",
                    },
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": "Status",
                      },
                      "ref": null,
                      "rendered": "Status",
                      "type": "th",
                    },
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": "Endre",
                      },
                      "ref": null,
                      "rendered": "Endre",
                      "type": "th",
                    },
                  ],
                  "type": "tr",
                },
                "type": "thead",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [],
                },
                "ref": null,
                "rendered": Array [],
                "type": "tbody",
              },
            ],
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
